routers/basket.js

```diff
@@ -0,0 +1,304 @@
+const { Op } = require('sequelize')
+const { Router } = require("express");
+const Baskets = require('../models').basket 
+const Products = require('../models').product 
+const Client = require('../models').client
+const BasketProducts = require('../models').basketProduct
+
+
+
+const router = new Router();
+
+router.get('/:id', async(req,res,next)=>{
+    try{
+    const id = req.params.id
+    const basket = await Baskets.findByPk(id,{
+        include: [Products]
+    })
+    const whatIsBasketProduct = await BasketProducts.findAll({
+        where: {
+            basketId: basket.id
+        }
+    })
+    
+   
+    const fullCart = whatIsBasketProduct.reduce((acc, p) => {
+        const pInAcc = acc.find(cp => cp.productId === p.productId)
+        if (!pInAcc) {
+        return [...acc, { productId: p.productId, quantity: 1 }]
+        } else {
+          const newAcc = acc.map(cp => cp.productId === p.productId ? { ...cp, quantity: cp.quantity + 1 } : cp)
+          return newAcc
+        }
+      }
+      , [])
+    
+    
+    const productIds = fullCart.map(p => p.productId)
+    const products = await Products.findAll({ where: { id: { [Op.in]: productIds } }})
+    const fullCartWithPrice = fullCart.map(cp => ({ ...cp, ...products.find(p => p.id === cp.productId).dataValues }))
+    
+    res.status(200).send(fullCartWithPrice)
+    } catch(e){
+        next(e)
+    }
+
+})
+
+router.get('/client/:id', async(req,res,next)=>{
+    try{
+    const id = req.params.id
+    const basket = await Baskets.findOne({
+        where: {
+        clientId : id,
+        },
+        include: [Products]
+    })
+   
+    const whatIsBasketProduct = await BasketProducts.findAll({
+        where: {
+            basketId: basket.id
+        }
+    })
+    
+   
+    const fullCart = whatIsBasketProduct.reduce((acc, p) => {
+        const pInAcc = acc.find(cp => cp.productId === p.productId)
+        if (!pInAcc) {
+        return [...acc, { productId: p.productId, quantity: 1 }]
+        } else {
+          const newAcc = acc.map(cp => cp.productId === p.productId ? { ...cp, quantity: cp.quantity + 1 } : cp)
+          return newAcc
+        }
+      }
+      , [])
```

Comment:
this logic that is repeated can go to a `utils` file on a function maybe called `buildCartFromBasket(array)`